prototypes/ScheduledMerges.hs

```diff
@@ -337,10 +337,10 @@ invariant (LSMContent _ levels ul) = do
     -- on the progress of the merges, since union merge credits are independent
     -- from the tables' regular level merges. However, we know that:
     --
-    -- * union merges are always binary
+    -- * union have exactly two other trees as inputs
     -- * level merges are non-trivial (at least two inputs)
-    -- * only the last input of a level merge can be a union merge or contain
-    --   a nested PendingMerge
+    -- * level merges consist of level-merge IncomingRuns and only up to one
+    --   arbitrary nested MergingTree
```

Comment:
You don't think more of this is worth catching in the type itself?

We could have PendingMerge enumerate more of these cases, capturing the structure for each. And we could still provide a view pattern to access it in the above simple representation (hiding the multiple cases) to keep most code simple.

Reply:
I had tried that, but didn't like that it made some code more noisy, despite it not caring about the merge type. A pattern synonym is a nice suggestion here. I like the result and will push it.